add_trailing_comma.py

```diff
@@ -92,7 +112,11 @@ def visit_Tuple(self, node):
         self.generic_visit(node)
 
     def visit_Call(self, node):
-        argnodes = node.args + node.keywords
+        # type: (ast.Call) -> None
+        argnodes = (
+            cast('List[ast.AST]', node.args) +
+            cast('List[ast.AST]', node.keywords)
+        )
```

Comment:
I'm surprised you need a type annotation here? what was it mad about?  does `# type: List[ast.AST]` help it enough?

Reply:
Without this:
```diff
diff --git a/add_trailing_comma.py b/add_trailing_comma.py
index a64b028..e7f67f6 100644
--- a/add_trailing_comma.py
+++ b/add_trailing_comma.py
@@ -8,7 +8,6 @@ import collections
 import io
 import sys
 import warnings
-from typing import cast
 from typing import Dict
 from typing import Generator
 from typing import List
@@ -113,10 +112,9 @@ class FindNodes(ast.NodeVisitor):
 
     def visit_Call(self, node):
         # type: (ast.Call) -> None
-        argnodes = (
-            cast('List[ast.AST]', node.args) +
-            cast('List[ast.AST]', node.keywords)
-        )
+        argnodes = node.args + node.keywords
+        reveal_type(node.args)
+        reveal_type(node.keywords)
         py2_starargs = getattr(node, 'starargs', None)
         if py2_starargs:  # pragma: no cover (<PY35)
             argnodes.append(py2_starargs)
```

yields

```console
$ pre-commit run mypy --all-files
mypy.....................................................................Failed
hookid: mypy

add_trailing_comma.py:115: error: Unsupported operand types for + ("List[expr]" and "List[keyword]")
add_trailing_comma.py:116: note: Revealed type is 'builtins.list[_ast.expr]'
add_trailing_comma.py:117: note: Revealed type is 'builtins.list[_ast.keyword]'
```

So I used the common ancestor 🤷‍♂ 

```pycon
>>> import ast
>>> ast.expr.mro()
[<class '_ast.expr'>, <class '_ast.AST'>, <class 'object'>]
>>> ast.keyword.mro()
[<class '_ast.keyword'>, <class '_ast.AST'>, <class 'object'>]
```

Reply:
hmmm this feels like a bug in mypy -- what if you add `# type: List[ast.AST]` does that make it happy without `cast`?

Reply:
I think I tried that but let me try again

Reply:
```diff
diff --git a/add_trailing_comma.py b/add_trailing_comma.py
index a64b028..f09e76a 100644
--- a/add_trailing_comma.py
+++ b/add_trailing_comma.py
@@ -8,7 +8,6 @@ import collections
 import io
 import sys
 import warnings
-from typing import cast
 from typing import Dict
 from typing import Generator
 from typing import List
@@ -113,10 +112,7 @@ class FindNodes(ast.NodeVisitor):
 
     def visit_Call(self, node):
         # type: (ast.Call) -> None
-        argnodes = (
-            cast('List[ast.AST]', node.args) +
-            cast('List[ast.AST]', node.keywords)
-        )
+        argnodes = node.args + node.keywords  # type: List[ast.AST]
         py2_starargs = getattr(node, 'starargs', None)
         if py2_starargs:  # pragma: no cover (<PY35)
             argnodes.append(py2_starargs)
```

```console
$ pre-commit run mypy --all-files
mypy.....................................................................Failed
hookid: mypy

add_trailing_comma.py:115: error: Incompatible types in assignment (expression has type "List[expr]", variable has type "List[AST]")
add_trailing_comma.py:115: note: "List" is invariant -- see http://mypy.readthedocs.io/en/latest/common_issues.html#variance
add_trailing_comma.py:115: note: Consider using "Sequence" instead, which is covariant
add_trailing_comma.py:115: error: Unsupported operand types for + ("List[expr]" and "List[keyword]")
```

Reply:
found a solution that doesn't require `cast` -- silly mypy

Reply:
awesome